cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/index.tsx

```diff
@@ -73,23 +79,54 @@ const ContentView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
     return null;
   }
 
-  const Comp = STEPS[activeStep].component;
+  const renderContentPage = () => {
+    const Comp = STEPS[activeStep].component;
+    const CompLabel = STEPS[activeStep].label;
+    switch (CompLabel) {
+      case 'Basic Plugin Information':
+        return (
+          <PluginInfoContext.Provider value={pluginInfoContextValue}>
+            <Comp />
+          </PluginInfoContext.Provider>
+        );
+      case 'Configuration Groups':
+        return (
+          <ConfigurationGroupContext.Provider value={configuratioGroupContextValue}>
+            <WidgetContext.Provider value={widgetContextValue}>
+              <Comp />
+            </WidgetContext.Provider>
+          </ConfigurationGroupContext.Provider>
+        );
+      case 'Output':
+        return (
+          <OutputContext.Provider value={outputContextValue}>
+            <Comp />
+          </OutputContext.Provider>
+        );
+      case 'Filters':
+        return (
+          <FilterContext.Provider value={filterContextValue}>
+            <Comp />
+          </FilterContext.Provider>
```

Comment:
you can do this from the STEPS definition instead of doing it here. I think this should remain generic

Reply:
addressed per our discussion.

Resolved.